### PR TITLE
[Merged by Bors] - Update `keyboard.rs` docs in `bevy_input`

### PR DESCRIPTION
**Pull request by KDecay**
# Objective

- Part of the splitting process of #3692.

## Solution

- Document `keyboard.rs` inside of `bevy_input`.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_input/src/keyboard.rs
Comment (alice-i-cecile) [inline crates/bevy_input/src/keyboard.rs:269]: Any context on what this is? I can't turn up anything in my searches.

File: crates/bevy_input/src/keyboard.rs
Comment (alice-i-cecile) [inline crates/bevy_input/src/keyboard.rs:271]: Same as `At`

File: crates/bevy_input/src/keyboard.rs
Comment (KDecay) [reply crates/bevy_input/src/keyboard.rs:269]: I suppose that this would be a button that just outputs `@` if that's a thing?

Comment (alice-i-cecile) [reply crates/bevy_input/src/keyboard.rs:269]: Yeah, that seems right.

File: crates/bevy_input/src/keyboard.rs
Comment (KDecay) [reply crates/bevy_input/src/keyboard.rs:271]: I also couldn't find anything on the internet. No idea.

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_input/src/keyboard.rs
Comment (alice-i-cecile) [reply crates/bevy_input/src/keyboard.rs:271]: Yeah, this appears to be `winit's` fault. I'll file an issue.

Comment (alice-i-cecile) [reply crates/bevy_input/src/keyboard.rs:271]: https://github.com/rust-windowing/winit/issues/2265

**Comment by alice-i-cecile** [comment]
@bevyengine/docs-team can I get another set of eyes on this?

**Comment by devil-ira** [comment]
Not a fan of "__ key, next to __". Seems unnecessary, and it assumes the reader has a 'standard' keyboard layout.
It works out for my desktop's keyboard but, it does not match my laptop's keyboard.

**Comment by alice-i-cecile** [comment]
Yeah, that's sensible. Perhaps "distinct from `Num0`" is more helpful?

File: crates/bevy_input/src/keyboard.rs
Comment (devil-ira) [inline crates/bevy_input/src/keyboard.rs:82]:
```suggestion
    /// The `0` key over the letters.
```
If they can find 1 through 9 they can probably find 0 :wink:, and this won't be true for all keyboard layouts.

Comment (alice-i-cecile) [reply crates/bevy_input/src/keyboard.rs:82]: Yep, "over the letters" is better.

File: crates/bevy_input/src/keyboard.rs
Comment (devil-ira) [inline crates/bevy_input/src/keyboard.rs:219]:
```suggestion
    /// The `Back` / `Backspace` key.
```
@alice-i-cecile I was actually talking about these. This one is actually incorrect for both my desktop and laptop keyboard. `\` sits in-between  `enter` and `backspace` for me. Here's an [image](https://tweakers.net/i/6_zaBrGnwQjzag8OAEZcX7LsUp8=/i/2002843126.jpeg) of my keyboard.

Comment (alice-i-cecile) [reply crates/bevy_input/src/keyboard.rs:219]: Yeah, same here for me. I like this better.

File: crates/bevy_input/src/keyboard.rs
Comment (IceSentry) [reply crates/bevy_input/src/keyboard.rs:82]: Even "over the letters" isn't true for some keyboards. Some people using tiny keyboards have the number keys on a different layer entirely. Or even ergonomic split keyboards might have the numbers somewhere else.

File: crates/bevy_input/src/keyboard.rs
Comment (IceSentry) [inline crates/bevy_input/src/keyboard.rs:53]: is the name of the resource missing here? What's a `bevy` resource?

File: crates/bevy_input/src/keyboard.rs
Comment (IceSentry) [inline crates/bevy_input/src/keyboard.rs:11]: Should this say that the event is _updated_ by the system? It's a bit confusing to me to only say it's read by a system.

Comment (IceSentry) [reply crates/bevy_input/src/keyboard.rs:11]: Right, it does get cleared by the system so _consumed_ does sounds better in that case.

File: crates/bevy_input/src/keyboard.rs
Comment (devil-ira) [reply crates/bevy_input/src/keyboard.rs:82]: Did some 'research' on google images.

Every Split keyboard I could find that had number keys, had them positioned over the letters.
40% keyboards indeed have the numbers on the top row on another layer, but I believe that someone using a niche keyboard will realize what is meant by 'over the letters'.

Comment (IceSentry) [reply crates/bevy_input/src/keyboard.rs:82]: Yeah, I should have been more specific, my friend has a split keyboard with blank keys and custom layers so the numbers could be anywhere but that's not really related to the fact that it's a split keyboard.

Either way, I still think that mentioning the position of the number keys relative to the letter is unnecessary, but I won't block the PR on that.

Comment (alice-i-cecile) [reply crates/bevy_input/src/keyboard.rs:82]: I think the critical idea is "not numpad", which should be conveyed somehow.

File: crates/bevy_input/src/keyboard.rs
Comment (KDecay) [reply crates/bevy_input/src/keyboard.rs:53]: No. A `bevy` resource is anything that can be used with `Res<Type>` inside of a system. This essentially tells you that we create a `bevy` resource using `KeyCode` as the generic `T` of `Input` creating the `Input<KeyCode>` resource that can be accessed using `Res<Input<KeyCode>>`.

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_input/src/keyboard.rs
Comment (IceSentry) [reply crates/bevy_input/src/keyboard.rs:53]: In that case, I think I would say "to create a `Res<Input<KeyCode>>`".

**Review by BurningBlueFox — APPROVED** [review]

**Pull request closed without merging**
